docs/decisions/0005-stripe-custom-actions.rst

```diff
@@ -0,0 +1,93 @@
+5. Stripe Custom Actions 
+========================
+
+
+Status
+------
+
+Draft
+
+.. After acceptance, becomes, for example:
+
+    Accepted (2023-02-01
+
+
+Context
+-------
+
+We have added Stripe as a payment processor suppported by frontend-app-payment.
+
+`Stripe's default payment architecture`_ does not involve a service's backend until after payment. This means a user can pay before the backend service can evaluate the user's billing details.
+
+Our `ecommerce` backend may deny checkout if a user's billing details are on a sanctions list. This is called the `SDN check`_. But for this, ecommerce must be involved before payment.
+
+`Stripe Custom Actions`_ is an alternative architecture that sends payments through a backend before payment. Stripe Custom Actions preserves PCI compliance by sending sensitive financial information to Stripe and not the backend. However, it does not process payment until the payment is confirmed by the backend and allows inspection of non-sensitive billing details by the backend prior to payment.
+
+Stripe Custom Actions also is similar to the architecture of Cybersource's Flex Microform SDK. Both architectures do not involve a pending state for baskets and involve the backend prior to payment. This similarity makes it easier to intially implement Stripe as `BasePaymentProcessor`_ in ecommerce.
+
+.. _Stripe's default payment architecture: https://stripe.com/docs/payments/accept-a-payment?platform=web&ui=elements#web-create-intent
+.. _ecommerce: https://github.com/openedx/ecommerce
+.. _SDN check: https://github.com/openedx/ecommerce/blob/master/ecommerce/extensions/payment/core/sdn.py#L29
+.. _BasePaymentProcessor: https://github.com/openedx/ecommerce/blob/7555353ae972563fd293558eea608ae6151c1186/ecommerce/extensions/payment/processors/__init__.py#L20 
+
+
+Decision
+--------
+
+We will use the architecture in the `Stripe Custom Actions`_ documentation for communication between frontend-app-payment, frontend-app-payment's backend (ecommerce) and Stripe.
+
+.. _Stripe Custom Actions: https://stripe.com/docs/payments/run-custom-actions-before-confirmation
+
+
+Consequences
+------------
+
+When frontend-app-payment loads, it will ask ecommerce for a Stripe ``client_secret`` using ecommerce's endpoint ``/bff/payment/v0/capture_context/``.
+
+When a user submits a purchase, frontend-app-payment will:
+
+#. Send the payment details to Stripe via ``stripe.updatePaymentIntent()``. (As opposed to executing payment via ``stripe.confirmPayment()``.)
```

Comment:
Worth adding that this method difference is due to CAB?

Reply:
Done.